src/model.cpp

```diff
@@ -127,9 +140,87 @@ std::string Model::doSerialisation(FORMATS format) const
 
         repr += serialiseUnits(format);
 
-        // Serialize components of the model, imported components have already been dealt with at this point.
+        // Serialise components of the model, imported components have already been dealt with at this point.
         for (size_t i = 0; i < componentCount(); ++i) {
             repr += getComponent(i)->serialise(format);
+            // Build unique variable equivalence pairs (VariableMap) for connections
+            ComponentPtr component = getComponent(i);
+            for (size_t j = 0; j < component->variableCount(); ++j) {
+                VariablePtr variable = component->getVariable(j);
+                if (variable->equivalentVariableCount() > 0) {
+                    for (size_t k = 0; k < variable->equivalentVariableCount(); ++k) {
+                        VariablePtr equivalentVariable = variable->getEquivalentVariable(k);
+                        VariablePair variablePair = std::make_pair(variable, equivalentVariable);
+                        VariablePair reciprocalVariablePair = std::make_pair(equivalentVariable, variable);
+                        bool pairFound = false;
+                        for (VariableMapIterator iter = variableMap.begin(); iter < variableMap.end(); ++iter) {
+                            if (*iter == variablePair || *iter == reciprocalVariablePair) {
```

Comment:
I haven't checked the coding style, but personally I prefer to make it (very) clear what we are testing. In other words, I would have:

```
...
if ((*iter == variablePair) || (*iter == reciprocalVariablePair)) {
...
```

Same elsewhere in the code (e.g. line 185 here).